### PR TITLE
Fix classes

### PR DESCRIPTION
**Pull request by Jleagle**

### CODE REVIEW

File: src/Html/TemplatedHtmlElement.php [hunk 1]
```diff
@@ -3,16 +3,16 @@
...
 {
   use TemplateLoaderTrait;
 
   /**
-   * @return SafeHtml|null
-   * @throws \Throwable
+   * @throws Throwable
    */
-  protected function _getContentForRender()
+  protected function _getContentForRender(): SafeHtml
```
Comment (TomK) [inline src/Html/TemplatedHtmlElement.php]: probably breaking change due to things extending it

**Review by TomK — CHANGES_REQUESTED** [review]

File: src/Html/HtmlAttributesTrait.php [hunk 2]
```diff
@@ -242,10 +241,14 @@ private function _removeClass(string $class)
   /**
    * Retrieve all classes set on the element
    *
-   * @return string[]
+   * @return array<string>
```
Comment (TomK) [inline src/Html/HtmlAttributesTrait.php]:
```suggestion
   * @return string[]
```
this was correct before. typed array notation is only really useful if you need to specify key type or multi-dimensional arrays

File: src/Html/Uri.php [hunk 3]
```diff
@@ -96,8 +96,8 @@ public function __toString()
       $auth = '';
       if(strlen($this->user) && strlen($this->pass))
       {
-        $auth = SafeHtml::escapeUri($this->user) . ':' .
-          SafeHtml::escapeUri($this->pass) . '@';
+        /** @noinspection PhpUndefinedMethodInspection */
```
Comment (TomK) [inline src/Html/Uri.php]: why do we need this?

File: src/Html/Uri.php [hunk 4]
```diff
@@ -139,22 +139,28 @@ public function __toString()
     return $prefix . $this->getPath() . $query . $fragment;
   }
 
-  public function getFragment()
+  public function getFragment(): string
```
Comment (TomK) [inline src/Html/Uri.php]:
```suggestion
  public function getFragment()
```

File: src/Html/Uri.php [hunk 5]
```diff
@@ -139,22 +139,28 @@ public function __toString()
...
+   * @return $this
+   */
   public function setFragment($fragment)
   {
     $this->fragment = $fragment;
     return $this;
   }
 
-  public function getPath()
+  public function getPath(): string
```
Comment (TomK) [inline src/Html/Uri.php]:
```suggestion
  public function getPath()
```

**Review by TomK — APPROVED** [review]

**Review by bajb — APPROVED** [review]

**Pull request merged**